StatsService.js

```diff
@@ -79,15 +78,20 @@ class StatsService {
    */
   subscribe(subscription) {
     this.subscribers.add(subscription);
+    this.start();
   }
   /**
    * Each subscription is unique and can be deleted.
+   * If there are no subscriptions stop polling the api.
    *
    * @param {Subscription} subscription
    * - This can be a ChartController's subscription function.
    */
   unsubscribe(subscription) {
-    this.subscribers = this.subscribers.delete(subscription);
+    this.subscribers.delete(subscription);
+    if(!this.subscribers.size) {
```

Comment:
Is this the right check? Are we sure it's not `this.subscribers.size === 0`?

Reply:
fixed it you are correct there could be edge cases such some overwriting the prototype for size that would lead to null or undefined resulting in false. thanks and good catch.